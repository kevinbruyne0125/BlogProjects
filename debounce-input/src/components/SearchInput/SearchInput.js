import React, { Component } from "react";

class SearchInput extends Component {
  render() {
    return (
      <form className="mb-4">
        <div className="flex flex-col mb-4 md:w-full">
          <label
            className="mb-2 uppercase font-bold text-lg text-grey-darkest"
            htmlFor="search-input"
          >
            Search input:
          </label>
          <input className="field" name="search" type="text" id="search" />
        </div>
      </form>
    );
  }
}

export default SearchInput;
