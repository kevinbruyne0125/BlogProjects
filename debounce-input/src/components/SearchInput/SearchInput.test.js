import React from "react";

import { shallow } from "enzyme";

import SearchInput from "./SearchInput";

describe("SearchInput Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SearchInput />);
    expect(wrapper).toMatchSnapshot();
  });
});
