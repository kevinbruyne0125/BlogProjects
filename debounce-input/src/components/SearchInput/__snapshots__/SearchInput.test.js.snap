// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flex flex-col mb-4 md:w-full"
>
        <label
                className="mb-2 uppercase font-bold text-lg text-grey-darkest"
                htmlFor="search-input"
        >
                Search input:
        </label>
        <input
                className="field"
                id="search"
                name="search"
                type="text"
        />
</div>,
      "className": "mb-4",
      "onChange": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="mb-2 uppercase font-bold text-lg text-grey-darkest"
            htmlFor="search-input"
>
            Search input:
</label>,
          <input
            className="field"
            id="search"
            name="search"
            type="text"
/>,
        ],
        "className": "flex flex-col mb-4 md:w-full",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search input:",
            "className": "mb-2 uppercase font-bold text-lg text-grey-darkest",
            "htmlFor": "search-input",
          },
          "ref": null,
          "rendered": "Search input:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "field",
            "id": "search",
            "name": "search",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flex flex-col mb-4 md:w-full"
>
          <label
                    className="mb-2 uppercase font-bold text-lg text-grey-darkest"
                    htmlFor="search-input"
          >
                    Search input:
          </label>
          <input
                    className="field"
                    id="search"
                    name="search"
                    type="text"
          />
</div>,
        "className": "mb-4",
        "onChange": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="mb-2 uppercase font-bold text-lg text-grey-darkest"
              htmlFor="search-input"
>
              Search input:
</label>,
            <input
              className="field"
              id="search"
              name="search"
              type="text"
/>,
          ],
          "className": "flex flex-col mb-4 md:w-full",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search input:",
              "className": "mb-2 uppercase font-bold text-lg text-grey-darkest",
              "htmlFor": "search-input",
            },
            "ref": null,
            "rendered": "Search input:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "field",
              "id": "search",
              "name": "search",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
